config/config.go

```diff
@@ -187,6 +195,28 @@ func (c *Config) GetLogger() *logrus.Logger {
 	return c.logger
 }
 
+func (c *Config) GetTracer() *tracing.Tracer {
+	if c.tracer == nil {
```

Comment:
Are there any stats on reliability and/or network performance, specifically wrt to memory use under high load? If it's minimal we can have this always enabled. If it isn't, it should be either enabled by default (but possible to disable) or disabled by default (maybe the better option).

Reply:
With the current setup, the `opentracing.GlobalTracer()` will be set to the [`NoopTracer`](https://github.com/opentracing/opentracing-go/blob/master/globaltracer.go#L4) if one does not provide any tracing config (_or if they provide invalid tracing config_) when booting up Hydra. It is disabled by default.

I use the `IsLoaded` helper later to determine as to whether to append the tracer middleware.

Reply:
Oh I see, so omitting `TRACING_PROVIDER` disables tracing?

Reply:
Exactly.

Reply:
Perfect

Reply:
With that said, people that configure tracing should really know what they are doing. The Google dapper paper is a great read - https://ai.google/research/pubs/pub36356

Reply:
Added tests to cover this.

Reply:
Added unit tests to cover above.